packages/edit-site/src/components/page-templates/index.js

```diff
@@ -315,6 +323,38 @@ export default function PageTemplates() {
 				elements: authors,
 				width: '1%',
 			},
+			{
+				header: __( 'Post types' ),
+				id: 'postTypes',
+				getValue: ( { item } ) => item.post_types,
+				render: ( { item } ) =>
+					item.post_types
+						?.map(
+							( postType ) => POST_TYPES[ postType ] || postType
+						)
+						?.join( ',' ),
+				elements: [
+					{ value: 'post', label: POST_TYPES.post },
+					{ value: 'page', label: POST_TYPES.page },
+				],
```

Comment:
Are these the only possible post types?

Reply:
No, they are not. There's a TODO note over there to look at this later, when the custom filtering was addressed.

Reply:
Pushed support for any registered post type that is viewable & declares editor support (as a proxy for template support, otherwise, media post type will be listed as well). https://github.com/WordPress/gutenberg/pull/62075/commits/35300bb6c21899af6775caba4b0a2eae8d2ff0c9 and https://github.com/WordPress/gutenberg/pull/62075/commits/18756b3321a7268d5d9cf39f83d2111efca268c4